connection.js

```diff
@@ -1750,30 +1762,34 @@ class Connection {
             case constants.deny:
                 this.respond(550, msg, () => {
                     self.msg_count.reject++;
+                    self.transaction.msg_status = 'rejected';
                     self.reset_transaction(() => self.resume());
                 });
                 break;
             case constants.denydisconnect:
                 this.respond(550, msg, () => {
                     self.msg_count.reject++;
+                    self.transaction.msg_status = 'rejected';
                     self.disconnect();
                 });
                 break;
             case constants.denysoft:
                 this.respond(450, msg, () => {
                     self.msg_count.tempfail++;
+                    self.transaction.msg_status = 'deferred';
                     self.reset_transaction(() => self.resume());
                 });
                 break;
             case constants.denysoftdisconnect:
                 this.respond(450, msg, () => {
                     self.msg_count.tempfail++;
+                    self.transaction.msg_status = 'deferred';
                     self.disconnect();
                 });
                 break;
             default:
                 outbound.send_email(this.transaction, (retval2, msg2) => {
-                    if (!msg2) msg2 = self.queue_msg(retval2, msg2);
+                    if (!msg2) msg2 = self.queue_msg(retval2, msg);
```

Comment:
Can you explain the purpose of this change?

Reply:
And shouldn't it also do the `|| 'Message Queued'` from above?

Reply:
I'm a bit perplexed. 1. Because `msg2` is `null` in that case and we'd like to use the message that queue plugin returned. 2. It shouldn't do because that's already done by `msg`.